File: js/beautiful_array.js
/**
 * @param {number} n
 * @return {number[]}
 */
 var beautifulArray = function(n) {
    let ans = [];
    
    ans.push(1);
    while (ans.length < n) {
        let temp = [];
        
        // odd 
        for (let i = 0; i < ans.length; i++) {
            if (ans[i] * 2 - 1  <= n) {
                temp.push(ans[i] * 2 - 1);
            }
        }
        
        // even
        for (let i = 0; i < ans.length; i++) {
            if (ans[i] * 2 <= n) {
                temp.push(ans[i] * 2);
            }
        }
        
        ans = temp;
    }
    
    return ans;
};

File: java/beautiful_array.java
class Solution {
    public int[] beautifulArray(int n) {
        ArrayList<Integer> ans = new ArrayList<>();
        
        ans.add(1);
        
        while (ans.size() < n) {
            ArrayList<Integer> temp = new ArrayList<>();
            // odd
            for (int i = 0; i < ans.size(); i++) {
                 if (ans.get(i) * 2 - 1 <= n) {
                     temp.add(ans.get(i) * 2 - 1);
                 }
            } 
               
            // even
            for (int i = 0; i < ans.size(); i++) {
                 if (ans.get(i) * 2 <= n) {
                     temp.add(ans.get(i) * 2);
                 } 
            }
            ans = temp;
        }
        
        int len = ans.size();
        int []result = new int[len];
        
        for (int i = 0; i < len; i++) {
            result[i] = ans.get(i);
        }
        
        return result;
    }
}